docs/topics/jvm/jvm-spring-boot-add-data-class.md

```diff
@@ -0,0 +1,154 @@
+[//]: # (title: Add a data class to Spring Boot project)
+[//]: # (description: Add a Kotlin data class to Spring Boot project.)
+
+<microformat>
+    <p>This is the second part of the <strong>Getting started with Spring Boot and Kotlin</strong> tutorial. Before proceeding, make sure you've completed previous steps:</p><br/>
+    <p><img src="icon-1-done.svg" width="20" alt="First step"/> <a href="jvm-create-project-with-spring-boot.md">Create a Spring Boot project with Kotlin</a><br/><img src="icon-2.svg" width="20" alt="Second step"/> <strong>Add a data class to the Spring Boot project</strong><br/><img src="icon-3-todo.svg" width="20" alt="Third step"/> Add database support for Spring Boot project<br/><img src="icon-4-todo.svg" width="20" alt="Fourth step"/> Use Spring Data CrudRepository for database access</p>
+</microformat>
+
+In this part of the tutorial, you'll add some more functionality to the application and discover more Kotlin language features, such as data classes.
+It requires changing the `MessageController` class to respond with a JSON document containing a collection of serialized objects.
+
+## Update your application
+
+1. In the `DemoApplication.kt` file, create a `Message` data class with two properties: `id` and `text`:
+
+    ```kotlin
+    data class Message(val id: String?, val text: String)
+    ```
+
+   `Message` class will be used for data transfer: a list of serialized `Message` objects will make up the JSON document that the controller is going to respond to the browser request.
+
+   <deflist collapsible="true">
+       <def title="Data classes – data class Message">
+          <p>The main purpose of <a href="data-classes.md">data classes</a> in Kotlin is to hold data. Such classes are marked with <code>data</code> keyword and some standard functionality and some utility functions are often mechanically derivable from the class structure.</p>
+          <p>In this example, you declared <code>Message</code> as a data class as its main purpose is to store the data.</p>
+       </def>
+       <def title="val and var properties">
+          <p><a href="properties.md">Properties in Kotlin</a> classes can be declared either as:</p>
+          <list>
+              <ul>
+                <li><i>mutable</i>, using the <code>var</code> keyword</li>
+                <li><i>read-only</i>, using the <code>val</code> keyword</li>
+              </ul>
+          </list>
+          <p>The <code>Message</code> class declares two properties using <code>val</code> keyword, the <code>id</code> and <code>text</code>.
+          The compiler will automatically generate the getters for both these properties.
+          However, it will not be possible to reassign the values of those properties after an instance of the <code>Message</code> class is created.
```

Comment:
```suggestion
          However, it will not be possible to reassign the values of these properties after an instance of the <code>Message</code> class is created.
```

Reply:
One sentence ago they were "these":)